File: sdk/go/provider/spam_dhsf_net.go
package provider

// spam.dhsf.net：mailinator 官方姊妹子域名，MX 指向 mail.mailinator.com。

// SpamDhsfNetGenerate 创建 spam.dhsf.net 临时邮箱
func SpamDhsfNetGenerate() (*CreatedMailbox, error) {
	local := randomStr(12)
	return &CreatedMailbox{
		Channel: "spam-dhsf-net",
		Email:   local + "@spam.dhsf.net",
	}, nil
}

// SpamDhsfNetGetEmails 读取邮件（复用 mailinator public API）
func SpamDhsfNetGetEmails(email string) ([]NormEmail, error) {
	return MailinatorGetEmails(email)
}

File: sdk/npm/src/providers/spam-dhsf-net.ts
import { InternalEmailInfo, Email, Channel } from "../types";
import * as mailinator from "./mailinator";

const CHANNEL: Channel = "spam-dhsf-net";
const DOMAIN = "spam.dhsf.net";

function randomString(length: number): string {
  const chars = "abcdefghijklmnopqrstuvwxyz0123456789";
  let result = "";
  for (let i = 0; i < length; i++) {
    result += chars[Math.floor(Math.random() * chars.length)];
  }
  return result;
}

/**
 * 创建 spam.dhsf.net 临时邮箱（mailinator 姊妹子域名，MX 指向 mail.mailinator.com）
 */
export async function generateEmail(): Promise<InternalEmailInfo> {
  const local = randomString(12);
  return { channel: CHANNEL, email: `${local}@${DOMAIN}` };
}

/**
 * 获取邮件列表（复用 mailinator public API）
 */
export async function getEmails(
  token: string,
  email: string,
): Promise<Email[]> {
  return mailinator.getEmails(token, email);
}
